### PR TITLE
Get keys

### PR DESCRIPTION
**Pull request by orthagonal**

### CODE REVIEW

File: plugins/store.js [hunk 1]
```diff
@@ -8,7 +8,19 @@ exports.register = function(server, options, next) {
   }
   server.decorate('server', 'store', {
     get: cache.get,
-    set: cache.set
+    set: cache.set,
+    scan(expression, done) {
+      const stream = cache.scanStream({ match: expression });
```
Comment (jgallen23) [inline plugins/store.js]: We need to support if redis host isn't set by looping through object

File: plugins/store.js [hunk 2]
```diff
@@ -8,7 +8,19 @@ exports.register = function(server, options, next) {
   }
   server.decorate('server', 'store', {
     get: cache.get,
-    set: cache.set
+    set: cache.set,
```
Comment (jgallen23) [inline plugins/store.js]: Get and set need to be wired up if no host

File: test/test.store.js [hunk 3]
```diff
@@ -36,3 +35,39 @@ tap.test('store plugin calls redis server', (t) => {
...
+      server.methods.getKeys('key', done);
+    },
+    verify1(scan1, done) {
+      t.equal(scan1.length, 1);
+      done();
+    },
+    scan2(setup, done) {
+      server.store.set('key1', 'value1');
+      server.store.set('key2', 'value2');
+      server.store.set('key3', 'value3');
```
Comment (jgallen23) [inline test/test.store.js:119]: You need to add something in here that doesn't match

Comment (jgallen23) [reply test/test.store.js:119]: Also, is set a syncronous call?

File: test/test.store.js [hunk 4]
```diff
@@ -36,3 +35,39 @@ tap.test('store plugin calls redis server', (t) => {
...
+      const rapptor = new Rapptor({ env: 'test' });
+      rapptor.start(done);
+    },
+    setup(rapptor, done) {
+      server = rapptor[0];
+      return done(null, server);
+    },
+    scan1(setup, done) {
+      t.equal(typeof server.store.scan, 'function', 'store.scan is registered');
+      server.methods.getKeys('key', done);
```
Comment (jgallen23) [inline test/test.store.js]: What makes this match something?

**Review by jgallen23 — CHANGES_REQUESTED** [review]

File: plugins/store.js [hunk 5]
```diff
@@ -3,12 +3,50 @@ const Redis = require('ioredis');
 exports.register = function(server, options, next) {
   const settings = server.settings.app;
   let cache = settings.cache;
-  if (settings.redis.host) {
-    cache = new Redis(settings.redis);
+  // use internal cache if no host given:
+  if (!settings.redis.host) {
+    server.decorate('server', 'store', {
+      get: cache.get,
```
Comment (jgallen23) [inline plugins/store.js]: This isn't right. Should just be a normal {} that we use just for testing

File: test/test.store.js (hunk 3 above)
Comment (jgallen23) [reply test/test.store.js:119]: What about this?

**Review by jgallen23 — CHANGES_REQUESTED** [review]

File: test/test.store.js (hunk 3 above)
Comment (orthagonal) [reply test/test.store.js:119]: Yeah set is synchronous in the ioredis docs, I think we can trick it into being async using ioredis's pipeline if needed.

Comment (jgallen23) [reply test/test.store.js:119]: https://github.com/luin/ioredis/issues/279 all cleared up

**Pull request merged**